Makefile

```diff
@@ -0,0 +1,84 @@
+.DEFAULT_GOAL := help
+
+.PHONY: help
+help:
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
+
+# -- variables --------------------------------------------------------------------------------------
+
+WARNINGS=RUSTDOCFLAGS="-D warnings"
+
+# -- linting --------------------------------------------------------------------------------------
+
+.PHONY: clippy
+clippy: ## Runs Clippy with configs
+	cargo +nightly clippy --workspace --all-targets --all-features -- -D warnings
+
+
+.PHONY: fix
+fix: ## Runs Fix with configs
+	cargo +nightly fix --allow-staged --allow-dirty --all-targets --all-features
+
+
+.PHONY: format
+format: ## Runs Format using nightly toolchain
+	cargo +nightly fmt --all
+
+
+.PHONY: format-check
+format-check: ## Runs Format using nightly toolchain but only in check mode
+	cargo +nightly fmt --all --check
+
+
+.PHONY: lint
+lint: format fix clippy ## Runs all linting tasks at once (Clippy, fixing, formatting)
+
+# --- docs ----------------------------------------------------------------------------------------
+
+.PHONY: doc
+doc: ## Generates & checks documentation 
+	$(WARNINGS) cargo doc --all-features --keep-going --release
+
+
+.PHONY: doc-serve
+doc-serve: ## Serves documentation site
+	./scripts/serve-doc-site.sh
+
+# --- testing -------------------------------------------------------------------------------------
+
+.PHONY: test
+test:  ## Runs all tests
+	cargo nextest run --all-features --workspace --no-capture
+
+# --- installing --------------------------------------------------------------------------------
+
+.PHONY: install-node
+install-node: ## Installs node with testing feature enabled
+	cargo install --features testing --path bin/node
+
+.PHONY: install-faucet
+install-faucet: ## Installs faucet with testing feature enabled
+	cargo install --features testing --path bin/faucet
```

Comment:
Should we have 2 versions of these commands? One for installing with `testing` feature and one without?

Reply:
Sure.